hack/newTargetGen/main.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2021 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+type component struct {
+	Name          string
+	UppercaseName string
+	FullCaps      string
+}
+
+func (a *component) replaceTemplates(filename, outputname string) {
+	tmp1, err := template.ParseFiles(filename)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	file, err := os.Create(outputname)
+	if err != nil {
+		log.Fatalf("failed creating file: %s", err)
+	}
+	defer file.Close()
+
+	err = tmp1.Execute(file, a)
+
+	if err != nil {
+		fmt.Println(err)
+	}
+
+}
+
+func main() {
+	temp := &component{}
+	// var capsName string
+
+	fmt.Print("Enter the target name: ")
+	fmt.Scanf("%s", &temp.UppercaseName)
+
+	temp.Name = strings.ToLower(temp.UppercaseName)
+	temp.FullCaps = strings.ToUpper(temp.UppercaseName)
+	// fmt.Print("Enter the ALL CAPS VERISON of the target name: ")
+	// fmt.Scanf("%s", &capsName)
+
+	// TODO add naming validation here
+
+	// make cmd folder
+	path := fmt.Sprintf("cmd/%s", temp.Name+"-adapter")
+	newpath := filepath.Join("../../", path)
```

Comment:
This is very fragile and assumes the caller is always inside `hack/newTargetGen/` when running the generator.
I think it would be saner to assume the _current directory_ is the base directory, and let the user select an alternative path using a flag.

Reply:
could i have a little help on this one? Im not exactly positive how to do this :/

Reply:
It's easier than your imagine :)

I'm sharing some snippets below, but in a nutshell you simply need to use `./` instead of `../../` (and ideally allow users to override that value if they run from a directory which is not the root of the repo).

1. Set a flag which default value is `./` (always represents the current working directory)

 
   https://github.com/triggermesh/triggermesh/blob/a490818b065d34b7528584936eb20f25b9ad0738/hack/rbac-check/main.go#L192-L193

1. Sanitize the user input (just a good practice)

 
   https://github.com/triggermesh/triggermesh/blob/a490818b065d34b7528584936eb20f25b9ad0738/hack/rbac-check/main.go#L199

1. Compose all the paths as a join of this base path and the target directory (like you already did)

    https://github.com/triggermesh/triggermesh/blob/a490818b065d34b7528584936eb20f25b9ad0738/hack/rbac-check/main.go#L96